### ISSUE TITLE
Poetry export git does not include commit hash

### ISSUE THREAD

**Issue by lindycoder** [python-poetry/poetry-plugin-export#71]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.  (PREVIEW)
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS Monterey 12.0.1
- **Poetry version**: 1.2.0a2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: default from poetry init -n

## Issue

Hello,

It seems exporting a git dependency does not include the `resolved_reference` from the poetry lock.  Instead it just references the branch which it's tracking.

```
poetry export
test-package @ git+ssh://git@github.com/lindycoder/poetry-git-export-issue.git@main ; python_version >= "3.10" and python_version < "4.0"
```

The goal of exporting is to install the exact versions installed in the poetry lock so...

### Expected
since my `poetry.lock` includes
```
[package.source]
type = "git"
url = "git@github.com:lindycoder/poetry-git-export-issue.git"
reference = "main"
resolved_reference = "640da7171839d4b7857728b321999ef7d5e171b2"
```
I would expect the export to be
```
poetry export
test-package @ git+ssh://git@github.com/lindycoder/poetry-git-export-issue.git@640da7171839d4b7857728b321999ef7d5e171b2 ; python_version >= "3.10" and python_version < "4.0"
```

### Reproduce
I prepared a very small package to test, locally you can

```
poetry init -n
poetry add git+ssh://git@github.com:lindycoder/poetry-git-export-issue.git#main
poetry export
```

**Comment by bobwhitelock** [comment]
Ran into the same issue, via https://github.com/moneymeets/python-poetry-buildpack - very surprising as an unrelated deploy caused a Git dependency to update to the latest commit on that branch, ignoring the lockfile, and breaking things in the deployed environment!

**Comment by davegaeddert** [comment]
It looks to me like this was fixed in https://github.com/python-poetry/poetry-plugin-export/commit/4670b0c5cb64693a695baf650432a0290501ac45, it just hasn't been released yet. @radoering is that something you could help with, or explain how it works? Thanks!

**Comment by radoering** [comment]
I'm planning to do a release just before the next poetry release. See python-poetry/poetry#8483 for the current status. Currently, I'm just waiting for a review of the race condition fix.

**Comment by radoering** [comment]
Resolved in poetry-plugin-export 1.6.0
